src/Types.hs

```diff
@@ -27,38 +27,63 @@ import qualified Data.Csv as CSV
 
 -- | Globally Unique ID (for reference in the database).  (FIXME: should we have different id types
 -- for different object types?)
-type GUID = Integer
+newtype GUID a = GUID Integer
+  deriving (Eq, Ord, Show, Read, Generic)
 
-data MetaInfo = MetaInfo
-    { _metaId        :: GUID
-    , _metaCreatedBy :: GUID
+data MetaInfo a = MetaInfo
+    { _metaId        :: GUID a
+    , _metaCreatedBy :: GUID User
     , _metaCreatedAt :: Timestamp
-    , _metaChangedBy :: GUID
+    , _metaChangedBy :: GUID User
     , _metaChangedAt :: Timestamp
     }
   deriving (Eq, Ord, Show, Read, Generic)
 
-newtype Article = Article { fromArticle :: [ST] }
+-- | Markdown content.
+newtype Document = Document { fromDocument :: ST }
```

Comment:
Document is the right name for the type.
However the constructor could reflect the fact that the document is in markdown format by being called Markdown, this then remains a newtype as long as we have only one kind of formatting.